presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java

```diff
@@ -45,10 +45,10 @@
     private final PlanNode root;
     private final Map<Symbol, Type> symbols;
     private final PartitioningHandle partitioning;
-    private final List<PlanNodeId> partitionedSources;
-    private final Set<PlanNodeId> partitionedSourcesSet;
+    private final List<PlanNodeId> tableScanSources;
+    private final Set<PlanNodeId> tableScanSourcesSet;
     private final List<Type> types;
-    private final Set<PlanNode> partitionedSourceNodes;
+    private final Set<PlanNode> tableScanSourceNodes;
```

Comment:
Let's make it `Set<TableScanNode>`

Reply:
In fact it is also used only in the `HttpRemoteTask`. Let's move it there.

This type of checks (`plan.getPartitionedSourceNodes().stream().anyMatch(TableScanNode.class::isInstance)`) can be replaced with `!plan.getSchedulingOrder().isEmpty()`. Or `plan.hasTableScans()` that is implemented as `!plan.getSchedulingOrder().isEmpty()`.